strax/context.py

```diff
@@ -1014,7 +1017,17 @@ def get_array(self, run_id: ty.Union[str, tuple, list],
                 max_workers=max_workers,
                 **kwargs)
             results = [x.data for x in source]
-        return np.concatenate(results)
+        
+        result = np.concatenate(results)
+        
+        for function in self.context_config['apply_data_function']:
```

Comment:
This function will be applied to all data_kinds/types which might not be desirable since it can be easily get forgotten while working in the same context. Maybe some dictionary would be here more desirable as strax.Option e.g. {'peak_basics': lambda res: res['area'] + 10}.

Reply:
Good suggestion. In the implementation I envisage for straxen this is desirable (want to check all datatypes). See https://github.com/XENONnT/straxen/pull/166 and more importantly https://github.com/XENONnT/analysiscode/blob/master/DAQ/remap_sectors/Software_remap.ipynb. 

I'd propose to also make the datatype the second arg of each function and do a regex in the function. 

>  ... since it can be easily get forgotten while working in the same context.

Perhaps it's good to stress that I think this is quite a big hammer that really should be expert only as the data that you load is not the data that is stored on disk (what you want for blinded data for example).

Reply:
@WenzDaniel I've updated this PR and updated the description above. This allows you to check it in the function you are feeding strax.

Reply:
Okay now I see, and now I also agree that we should apply this kind of correction in accumulate.